File: 14Feb2024.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int n = nums.length;
        int[] ans = new int[n];
        for (int i = 0, j = 0, k = 0; i < n; i++) {
            if (i % 2 == 0) {
                while (j < n && nums[j] < 0)
                    j++;
                ans[i] = nums[j++];
            } else {
                while (k < n && nums[k] > 0)
                    k++;
                ans[i] = nums[k++];
            }
        }
        return ans;
    }
}

File: 14Feb2024.py
from ast import List


class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = [0]*n
        j,k=0,0
        for i in range(n) :
            if(i%2==0) :
                while(j<n and nums[j]<0) :
                    j+=1
                ans[i]=nums[j]
                j+=1
            else :
                while(k<n and nums[k]>0) :
                    k+=1
                ans[i]=nums[k]
                k+=1
        
        return ans